
## wf_ngs_pipeline

### Inputs

#### Required

  * `config` (File, **required**)
  * `dataDir` (File, **required**)
  * `fastq_screen_configuration` (File, **required**)
  * `fastq_screen_contaminants` (File, **required**)
  * `genbankFile` (File, **required**)
  * `genome` (String, **required**)
  * `intervals` (File, **required**)
  * `min_allele_fraction` (Float, **required**)
  * `min_median_read_position` (Int, **required**)
  * `min_reads_per_strand` (Int, **required**)
  * `read1` (Array[File]+, **required**)
  * `read2` (Array[File]+, **required**)
  * `samplename` (String, **required**)

#### Optional

  * `task_bbduk.contamination` (File?)
  * `task_collect_wgs_metrics.bed` (File?)
  * `task_fastqc.adapters` (File?)
  * `task_fastqc.contaminants` (File?)
  * `task_fastqc.limits` (File?)
  * `task_snpEff.upDownStreamLen` (Int?)
  * `wf_collect_targeted_pcr_metrics.task_mark_duplicates.marked_bam` (String?)
  * `wf_minimap2.Indexing.splitIndex` (Int?)
  * `wf_minimap2.Mapping.howToFindGTAG` (String?)
  * `wf_minimap2.Mapping.matchingScore` (Int?)
  * `wf_minimap2.Mapping.maxFragmentLength` (Int?)
  * `wf_minimap2.Mapping.maxIntronLength` (Int?)
  * `wf_minimap2.Mapping.mismatchPenalty` (Int?)
  * `wf_minimap2.Mapping.retainMaxSecondaryAlignments` (Int?)

#### Defaults

  * `output_vcf_name` (String, default="all_variants.vcf")
  * `threads` (Int, default=1)
  * `task_bbduk.disk_size` (Int, default=100)
  * `task_bbduk.docker` (String, default="staphb/bbtools:39.01")
  * `task_bbduk.keep` (Boolean, default=true)
  * `task_bbduk.memory` (String, default="8GB")
  * `task_bbduk.number_of_retries` (Int, default=1)
  * `task_bbduk.threads` (Int, default=1)
  * `task_collect_multiple_metrics.docker` (String, default="broadinstitute/gatk:4.5.0.0")
  * `task_collect_multiple_metrics.memory` (String, default="8GB")
  * `task_collect_multiple_metrics.outputBasename` (String, default="multiple_metrics")
  * `task_collect_wgs_metrics.coverage_cap` (Int, default=250)
  * `task_collect_wgs_metrics.docker` (String, default="broadinstitute/gatk:4.5.0.0")
  * `task_collect_wgs_metrics.memory` (String, default="8GB")
  * `task_collect_wgs_metrics.minBaseQuality` (Int, default=20)
  * `task_collect_wgs_metrics.minMappingQuality` (Int, default=20)
  * `task_collect_wgs_metrics.outputFile` (String, default="collect_wgs_metrics.txt")
  * `task_collect_wgs_metrics.read_length` (Int, default=150)
  * `task_collect_wgs_metrics.sample_size` (Int, default=10000)
  * `task_collect_wgs_metrics.sensitivityFile` (String, default="collect_wgs_sensitivity_metrics.txt")
  * `task_collect_wgs_metrics.use_fast_algorithm` (Boolean, default=true)
  * `task_concat_2_vcfs.docker` (String, default="staphb/bcftools:1.17")
  * `task_create_sequence_dictionary.docker` (String, default="broadinstitute/gatk:4.5.0.0")
  * `task_create_sequence_dictionary.memory` (String, default="8GB")
  * `task_delly.docker` (String, default="dbest/delly:v1.0.0")
  * `task_delly.memory` (String, default="32GB")
  * `task_delly.svType` (String, default="DEL")
  * `task_depth_of_coverage.docker` (String, default="broadinstitute/gatk:4.5.0.0")
  * `task_depth_of_coverage.lower_coverage` (Int, default=10)
  * `task_depth_of_coverage.memory` (String, default="8GB")
  * `task_depth_of_coverage.min_base_quality` (Int, default=20)
  * `task_depth_of_coverage.outputPrefix` (String, default="depth_of_coverage")
  * `task_fastq_screen.aligner` (String, default="bwa")
  * `task_fastq_screen.docker` (String, default="dbest/fastq_screen:v0.15.3")
  * `task_fastq_screen.memory` (String, default="10GB")
  * `task_fastq_screen.subset` (Int, default=100000)
  * `task_fastq_screen.threads` (Int, default=1)
  * `task_fastqc.docker` (String, default="staphb/fastqc:0.12.1")
  * `task_fastqc.memory` (String, default="8GB")
  * `task_fastqc.threads` (Int, default=1)
  * `task_multiqc.docker` (String, default="multiqc/multiqc:v1.21")
  * `task_multiqc.memory` (String, default="8GB")
  * `task_snpEff.csvStats` (String, default="snpEff_summary.csv")
  * `task_snpEff.docker` (String, default="dbest/snpeff:v5.2a")
  * `task_snpEff.hgvs` (Boolean, default=true)
  * `task_snpEff.lof` (Boolean, default=true)
  * `task_snpEff.memory` (String, default="9G")
  * `task_snpEff.noDownstream` (Boolean, default=false)
  * `task_snpEff.noIntergenic` (Boolean, default=false)
  * `task_snpEff.noShiftHgvs` (Boolean, default=false)
  * `task_snpEff.outputPath` (String, default="./snpeff.vcf")
  * `task_snpEff.stats` (String, default="snpEff_summary.html")
  * `task_trimmomatic.disk_size` (Int, default=100)
  * `task_trimmomatic.docker` (String, default="staphb/trimmomatic:0.39")
  * `task_trimmomatic.memory` (String, default="8GB")
  * `task_trimmomatic.threads` (Int, default=4)
  * `task_trimmomatic.trimmomatic_minlen` (Int, default=40)
  * `task_trimmomatic.trimmomatic_quality_trim_score` (Int, default=15)
  * `task_trimmomatic.trimmomatic_window_size` (Int, default=4)
  * `wf_gatk.dockerImage` (String, default="broadinstitute/gatk:4.2.6.1")
  * `wf_gatk.javaXmx` (String, default="8G")
  * `wf_gatk.memory` (String, default="9G")
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.clip_overlapping_reads` (Boolean, default=true)
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.coverage_cap` (Int, default=250)
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.docker` (String, default="broadinstitute/gatk:4.5.0.0")
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.memory` (String, default="8GB")
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.minBaseQuality` (Int, default=20)
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.minMappingQuality` (Int, default=20)
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.outputMetrics` (String, default="collect_targeted_pcr_metrics.txt")
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.sample_size` (Int, default=10000)
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.sensitivityFile` (String, default="collect_targeted_pcr_sensitivity_metrics.txt")
  * `wf_collect_targeted_pcr_metrics.task_collect_targeted_pcr_metrics.target_coverage_file` (String, default="collect_targeted_pcr_target_coverage.txt")
  * `wf_collect_targeted_pcr_metrics.task_mark_duplicates.do_add_pg_tag_to_read` (Boolean, default=false)
  * `wf_collect_targeted_pcr_metrics.task_mark_duplicates.do_remove_duplicates` (Boolean, default=false)
  * `wf_collect_targeted_pcr_metrics.task_mark_duplicates.do_remove_sequencing_duplicates` (Boolean, default=false)
  * `wf_collect_targeted_pcr_metrics.task_mark_duplicates.docker` (String, default="broadinstitute/gatk:4.5.0.0")
  * `wf_collect_targeted_pcr_metrics.task_mark_duplicates.memory` (String, default="8GB")
  * `wf_collect_targeted_pcr_metrics.task_mark_duplicates.metrics_txt` (String, default="mark_duplicates_metrics.txt")
  * `wf_minimap2.Indexing.dockerImage` (String, default="staphb/minimap2:2.25")
  * `wf_minimap2.Indexing.kmerSize` (Int, default=15)
  * `wf_minimap2.Indexing.memory` (String, default="4G")
  * `wf_minimap2.Indexing.minimizerWindowSize` (Int, default=10)
  * `wf_minimap2.Indexing.timeMinutes` (Int, default=10)
  * `wf_minimap2.Indexing.useHomopolymerCompressedKmer` (Boolean, default=false)
  * `wf_minimap2.Mapping.dockerImage` (String, default="staphb/minimap2:2.25")
  * `wf_minimap2.Mapping.kmerSize` (Int, default=15)
  * `wf_minimap2.Mapping.memory` (String, default="30G")
  * `wf_minimap2.Mapping.secondaryAlignment` (Boolean, default=false)
  * `wf_minimap2.Mapping.skipSelfAndDualMappings` (Boolean, default=false)
  * `wf_minimap2.Mapping.softClippingForSupplementaryAlignments` (Boolean, default=true)
  * `wf_minimap2.Mapping.timeMinutes` (Int, default=1 + ceil((size(queryFile1,"G") * 200 / cores)))
  * `wf_minimap2.Mapping.writeLongCigar` (Boolean, default=true)

### Outputs

  * `fastq_screen_html` (File?)
  * `fastq_screen_txt` (File?)
  * `fastq_screen_tagged` (File?)
  * `fastq_screen_tagged_filter` (File?)
  * `forwardHtml` (File)
  * `reverseHtml` (File)
  * `forwardZip` (File)
  * `reverseZip` (File)
  * `forwardSummary` (File)
  * `reverseSummary` (File)
  * `forwardData` (File)
  * `reverseData` (File)
  * `trim_err` (File?)
  * `trim_stats` (File?)
  * `adapter_stats` (File?)
  * `phiX_stats` (File?)
  * `bam` (File?)
  * `bai` (File?)
  * `vcf` (File?)
  * `vcfIndex` (File?)
  * `vcfStats` (File?)
  * `vcfAligned` (File?)
  * `vcfAlignedIndex` (File?)
  * `vcfFiltered` (File?)
  * `vcfFilteredIndex` (File?)
  * `vcfFilteredStats` (File?)
  * `dellyVcf` (File?)
  * `vcf_concatenated` (File?)
  * `vcfAnnotated` (File?)
  * `multiple_metrics_outputs` (Array[File]?)
  * `depth_of_coverage_outputs` (Array[File]?)
  * `collect_wgs_output_metrics` (File?)
  * `collect_targeted_pcr_metrics` (File?)
  * `multiqc_report` (File?)
